app/Espo/Core/SelectManagers/Base.php

```diff
@@ -858,6 +867,17 @@ public function getSelectParams(array $params, $withAcl = false, $checkWherePerm
 
         $this->q($params, $result);
 
+        // check if entity has hasArchive activated
+        if ($this->metadata->get("scopes")[$this->entityType]['hasArchive']) {
```

Comment:
$this->metadata->get("scopes")[$this->entityType]['hasArchive'] should be changed to

$this->metadata->get(['scopes', $this->entityType, 'hasArchive'])

Reply:
OK